subsys/bluetooth/shell/bt.c

```diff
@@ -1330,10 +1455,9 @@ SHELL_CREATE_STATIC_SUBCMD_SET(bt_cmds) {
 	SHELL_CMD_ARG(name, NULL, "[name]", cmd_name, 1, 1),
 	SHELL_CMD_ARG(scan, NULL,
 		      "<value: on, passive, off> <dup filter: dups, nodups>",
-		      cmd_scan, 3, 0),
-	SHELL_CMD_ARG(advertise, NULL,
-		      "<type: off, on, scan, nconn> <mode: discov, non_discov>",
-		      cmd_advertise, 3, 0),
+		      cmd_scan, 1, 1),
+	SHELL_CMD_ARG(advertise, &sub_bt_advertise_cmds, "Control advertising",
+		      cmd_empty, 1, 0),
```

Comment:
Not sure why you changed to cmd_empty, are you planning on reusing this? Otherwise Id keep it as cmd_advertise, we could also perhaps pass NULL as command handler and then treat it as empty command handler in shell core.

Reply:
fixed